src/charm.py

```diff
@@ -143,6 +148,44 @@ def snap_path(self) -> Optional[str]:
 
         return self._snap_path
 
+    @property
+    def snap_channel(self) -> str:
+        """Get the channel for exporter snap.
+
+        The channel is determined by the controller version that the charm is deployed
+        under. In case the controller version is equal or higher than 2.6 and less
+        than 2.9, the snap channel is set to 2.8/stable. In case controller's major and
+        minor version match with 2.9, the snap channel is set to 2.9/stable. Otherwise,
+        raise ControllerIncompatibleError exception.
+        """
+        controller_version = self.get_controller_version()
+
+        if controller_version < version.parse("2.6"):
+            raise ControllerIncompatibleError(
+                f"Juju controller version {str(controller_version)} is too old "
+                + "and not supported. Please consider make an upgrade.",
+            )
+
+        if controller_version >= version.parse("3.0"):
+            raise ControllerIncompatibleError("Juju controller 3.x is not yet supported")
+
+        if controller_version < version.parse("2.9"):
```

Comment:
supernit: please reorder the comparisons from older to newer.
alternatively: comparing controller_version.major / minor may be a tiny bit more readable. E.g.
```
if controller_version.major == 2:
    if controller_version.minor <= 6:
        raise ...
    elif controller_version.minor in [7,8]:
        return "2.8/stable"
    elif controller_version.minor == 9:
        return "2.9/stable"

raise ControllerIncompatibleError
```

Reply:
I have refactored the version checking per your suggestion. Could you please take another look?